test/iidm/util/TerminalFinderTest.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <boost/test/unit_test.hpp>
+
+#include <powsybl/iidm/Bus.hpp>
+#include <powsybl/iidm/Network.hpp>
+#include <powsybl/iidm/Terminal.hpp>
+#include <powsybl/iidm/VoltageLevel.hpp>
+#include <powsybl/iidm/util/TerminalFinder.hpp>
+#include <powsybl/network/EurostagFactory.hpp>
+#include <powsybl/network/FictitiousSwitchFactory.hpp>
+
+namespace powsybl {
+
+namespace iidm {
+
+namespace util {
+
+BOOST_AUTO_TEST_SUITE(TerminalFinderTestSuite)
+
+BOOST_AUTO_TEST_CASE(testBbsTerminal) {
+    Network network = powsybl::network::FictitiousSwitchFactory::create();
+    VoltageLevel& vlN = network.getVoltageLevel("N");
+    stdcxx::Reference<Terminal> optTerminalN0 = vlN.getNodeBreakerView().getOptionalTerminal(0);
+    BOOST_CHECK(static_cast<bool>(optTerminalN0));
+
+    const auto& terminalStream = optTerminalN0.get().getBusBreakerView().getBus().get().getConnectedTerminals();
+    stdcxx::Reference<Terminal> bestOptTerminal = util::TerminalFinder::getDefault().find(terminalStream);
+    BOOST_CHECK(static_cast<bool>(bestOptTerminal));
+
+    Terminal& bestTerminal = bestOptTerminal.get();
+    BOOST_CHECK_EQUAL(ConnectableType::BUSBAR_SECTION, bestTerminal.getConnectable().get().getType());
+    BOOST_CHECK_EQUAL("N_0", bestTerminal.getBusBreakerView().getBus().get().getId());
+    BOOST_CHECK_EQUAL("N_0", bestTerminal.getBusView().getBus().get().getId());
+}
+
+BOOST_AUTO_TEST_CASE(testNoTerminal) {
+    Network network = powsybl::network::FictitiousSwitchFactory::create();
+    VoltageLevel& vlN = network.getVoltageLevel("N");
+
+    util::TerminalFinder slackTerminalFinder = util::TerminalFinder::getDefault();
```

Comment:
Not sure TerminalFinder should be copiable too

Reply:
Why ? The comparator is the only attribute and is copiable.

Reply:
Because this can be considered as a singleton that is not copyable by definition

Reply:
Ok